src/main/java/com/input/Configuration.java

```diff
@@ -11,11 +27,196 @@ public class Configuration
+                    ErrorHandler.logError("Name of data element is missing");
+                    return null;
+                }
+
+                dataName = elemDataID.getAttribute("name");         // Get dataID object attribute out of node
+
+                tempType = inferType(elemDataID.getAttribute("type"));              // Get dataID object attribute out of node
+
+                if (tempType == null){
+                    ErrorHandler.logError("Data type is missing");
+                    return null;
+                }
+
+                tempDataIDMap.put(dataName, new DataID(dataName, tempType));          // Place dataID object in dataIDList
+                this.dataIDList.add(new DataID(dataName, tempType));  // Should DataID objects have a attribute for whether it is a list or not
+            } else {
+                ErrorHandler.logError("XML file is structured improperly");
+                return null;
+            }
+        }
+        return tempDataIDMap;
+    }
+
+    /**
+     * Parse all event nodes from a given NodeList of event nodes and a map of DataID objects. This private method must
+     * execute after parseDataIDNodes() completes. If an error was detected in eventNodeList, the error will be logged,
+     * and the method will return null.
+     * @param eventNodeList - a NodeList of event nodes that are the parent of DataID nodes whose data type is elaborated
+     *                      in nodes from dataIDMap
+     * @param dataIDMap - a HashMap object containing dataID objects used to complete the DataID nodes for each event
+     * @return - boolean value that is true if the parsing was successful and false otherwise
+     */
+    private boolean parseEventNodes(NodeList eventNodeList, HashMap<String, DataID> dataIDMap){
+        ArrayList<DataID> curDataIDs = new ArrayList<>();
+        for (int i = 0; i < eventNodeList.getLength(); i++){
+            Node curNode = eventNodeList.item(i);
+            NodeList dataIDNodes = curNode.getChildNodes();
+            curDataIDs.clear();
+
+            if (curNode.getNodeType() == curNode.ELEMENT_NODE) {
+                Element elemEvent = (Element) curNode;
+                String eventName;
+
+                if (!elemEvent.hasAttribute("name")){
+                    ErrorHandler.logError("Event has no appropriate name.");
+                    return false;
+                }
+                eventName = elemEvent.getAttribute("name");
+
+                for (int j = 0; j < dataIDNodes.getLength(); j++){
+                    Node dataNode = dataIDNodes.item(j);
+
+                    if (dataNode.getNodeType() == curNode.ELEMENT_NODE){
+                        Element elemData = (Element) dataNode;
+
+                        String dataName = elemData.getAttribute("name");    // Get Event object attributes out of node and temp hashmap
```

Comment:
Other places you check that the "name" attribute exists, is there a reason you do not check here? 
If the attribute is missing, I believe it either returns an empty string or null. Documentation can tell us.